### PR TITLE
py-your: new package

### PR DESCRIPTION
**Pull request by aweaver1fandm**
Spack package recipe for YOUR, Your Unified Reader.  YOUR processes pulsar data in different formats.

Output below from spack install py-your
spack install py-your
[+] /usr (external glibc-2.28-oj2wjfl2ao5inhfz4qehw6hlck2hizvp)
[+] /opt/apps/spack/gcc-runtime-8.5.0-5k6kvi5
[+] /opt/apps/spack/libxcrypt-4.4.35-zigqpjo
[+] /opt/apps/spack/ncurses-6.4-xbvwv2w
[+] /opt/apps/spack/erfa-2.0.0-4qkta2n
[+] /opt/apps/spack/zlib-ng-2.1.6-ccn5qny
[+] /opt/apps/spack/pcre-8.45-33dfhul
[+] /opt/apps/spack/libpciaccess-0.17-2qdxdjo
[+] /opt/apps/spack/libmd-1.0.4-zbdiprt
[+] /opt/apps/spack/qhull-2020.2-klc7ewb
[+] /opt/apps/spack/bzip2-1.0.8-t65bq3t
[+] /opt/apps/spack/xz-5.4.6-axoznvt
[+] /opt/apps/spack/alsa-lib-1.2.3.2-a7icjdy
[+] /opt/apps/spack/zstd-1.5.6-nyk6gt6
[+] /opt/apps/spack/libffi-3.4.6-ibucrfe
[+] /opt/apps/spack/yasm-1.3.0-v4etmmm
[+] /opt/apps/spack/openblas-0.3.26-pfyk2vi
[+] /opt/apps/spack/pkgconf-1.9.5-ckjdqjm
[+] /opt/apps/spack/wcslib-7.3-zvcqq7o
[+] /opt/apps/spack/libiconv-1.17-jskazis
[+] /opt/apps/spack/unzip-6.0-mqftjtp
[+] /opt/apps/spack/libjpeg-turbo-3.0.0-vjvivme
[+] /opt/apps/spack/readline-8.2-2ys6ede
[+] /opt/apps/spack/openssl-3.2.1-4lqdgni
[+] /opt/apps/spack/pigz-2.8-rx263bp
[+] /opt/apps/spack/libpng-1.6.39-kiuku4y
[+] /opt/apps/spack/libbsd-0.12.1-njt5grs
[+] /opt/apps/spack/swig-4.0.2-fortran-z3sbnze
[+] /opt/apps/spack/binutils-2.42-vkjcvfr
[+] /opt/apps/spack/util-linux-uuid-2.38.1-w3kgjq3
[+] /opt/apps/spack/hdf5-1.14.3-sbuiw6q
[+] /opt/apps/spack/libedit-3.1-20230828-676jwbd
[+] /opt/apps/spack/nghttp2-1.57.0-u72gxms
[+] /opt/apps/spack/ffmpeg-6.1.1-2vhrbda
[+] /opt/apps/spack/libxml2-2.10.3-37klvxv
[+] /opt/apps/spack/gdbm-1.23-cylmqwx
[+] /opt/apps/spack/sqlite-3.43.2-axuxulg
[+] /opt/apps/spack/tar-1.34-wjzs4wj
[+] /opt/apps/spack/freetype-2.13.2-in4taxi
[+] /opt/apps/spack/expat-2.6.2-7kfe3hb
[+] /opt/apps/spack/curl-8.6.0-gpzsr3p
[+] /opt/apps/spack/hwloc-2.9.1-fhoz6al
[+] /opt/apps/spack/gettext-0.22.4-zjsp346
[+] /opt/apps/spack/lua-5.3.6-47356ia
[+] /opt/apps/spack/cfitsio-3.49-mmy3dbr
[+] /opt/apps/spack/python-3.10.13-fz7fymx
[+] /opt/apps/spack/elfutils-0.190-uswzaiw
[+] /opt/apps/spack/py-pytz-2023.3-ojdlhrd
[+] /opt/apps/spack/py-cycler-0.11.0-b7mjvv7
[+] /opt/apps/spack/py-pip-23.0-lxkcvby
[+] /opt/apps/spack/python-venv-1.0-2cz5c3s
[+] /opt/apps/spack/py-numpy-1.26.4-t5acjcv
[+] /opt/apps/spack/llvm-14.0.6-3nosumn
[+] /opt/apps/spack/py-packaging-23.1-wkeyuk6
[+] /opt/apps/spack/py-six-1.16.0-iv6iv3q
[+] /opt/apps/spack/py-pybind11-2.12.0-5rvupjv
[+] /opt/apps/spack/py-setuptools-69.2.0-3do76jw
[+] /opt/apps/spack/py-pyparsing-3.1.2-fq6imlt
[+] /opt/apps/spack/py-wheel-0.41.2-brm3k3h
[+] /opt/apps/spack/py-llvmlite-0.41.1-qom3l5h
[+] /opt/apps/spack/py-astropy-4.0.1.post1-xbojixg
[+] /opt/apps/spack/py-python-dateutil-2.8.2-kzdfskc
[+] /opt/apps/spack/py-numexpr-2.8.4-fc5xc5s
[+] /opt/apps/spack/py-h5py-3.11.0-y6drk3j
[+] /opt/apps/spack/py-tifffile-2023.8.30-oof4try
[+] /opt/apps/spack/py-pygments-2.16.1-stgrccl
[+] /opt/apps/spack/py-mdurl-0.1.2-nqk43ry
[+] /opt/apps/spack/py-scipy-1.13.0-vxjgfov
[+] /opt/apps/spack/py-contourpy-1.0.7-jg5lhss
[+] /opt/apps/spack/py-pillow-10.3.0-ijh2cju
[+] /opt/apps/spack/py-kiwisolver-1.4.5-vdh5sq5
[+] /opt/apps/spack/py-fonttools-4.39.4-x5ydbox
[+] /opt/apps/spack/py-bottleneck-1.3.7-ztsm4lg
[+] /opt/apps/spack/py-lazy-loader-0.4-k7hgvka
[+] /opt/apps/spack/py-numba-0.58.1-hzvrjwj
[+] /opt/apps/spack/py-markdown-it-py-3.0.0-l4p4qv5
[+] /opt/apps/spack/py-imageio-2.34.0-z5hu4yc
[+] /opt/apps/spack/py-matplotlib-3.8.4-azq2fzm
[+] /opt/apps/spack/py-pandas-1.5.3-p3gnh6t
[+] /opt/apps/spack/py-rich-13.4.2-okhgwwz
[+] /opt/apps/spack/py-networkx-3.1-b54br7r
[+] /opt/apps/spack/py-scikit-image-0.23.2-cvyzh3t
==> Installing py-your-0.6.7-djfzsn2lutp24ik6wrk6tjx5f7hil76x [83/83]
==> No binary for py-your-0.6.7-djfzsn2lutp24ik6wrk6tjx5f7hil76x found: installing from source
==> Fetching https://github.com/thepetabyteproject/your/archive/refs/tags/0.6.7.tar.gz
==> No patches needed for py-your
==> py-your: Executing phase: 'install'
==> py-your: Successfully installed py-your-0.6.7-djfzsn2lutp24ik6wrk6tjx5f7hil76x
  Stage: 1.43s.  Install: 0.99s.  Post-install: 0.12s.  Total: 3.12s

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-your/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    depends_on("py-h5py@2.10:", type=("build", "run"))
+    depends_on("py-scikit-image@0.14.2:", type=("build", "run"))
+    depends_on("py-scipy@1.3:", type=("build", "run"))
+    depends_on("py-numba@0.48:", type=("build", "run"))
+    depends_on("py-pandas@1.0.3:", type=("build", "run"))
+    depends_on("py-rich@8:", type=("build", "run"))
+
+    def setup_run_environment(self, env):
+        env.prepend_path("PATH", self.prefix.bin)
+        env.prepend_path("PATH", self.spec["python"].prefix.bin)
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/py-your/package.py]: I don't think these should be necessary.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/py-your/package.py]: AFAICT The core doesn't explicitly set these variables and there are 131 packages that do.

**Comment by aweaver1fandm** [comment]
This package has several .py files that are executable on their own (e.g.
your_candmaker.py) that wouldn't be available if I don't include the PATH
modification

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed the homepage and version sha256.

**Review by tldahlgren — DISMISSED** [review]
Confirmed the dependencies from the `requirements.txt` file.

File: var/spack/repos/builtin/packages/py-your/package.py (hunk 1 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/py-your/package.py]: Yeah, but I don't see `py-pip` doing this to ensure the `pip` command is in the `PATH`, or `py-tabulate` for the `tabulate` command (to take a much smaller package in terms of loc in `package.py`). I don't think this is necessary.

```
18:43:39 wdconinc@menelaos ~/git/spack (develop *$%>) $ which tabulate
18:43:46 wdconinc@menelaos ~/git/spack (develop *$%>) $ spack load py-tabulate/6xgmssa
18:43:57 wdconinc@menelaos ~/git/spack (develop *$%>) $ which tabulate
/opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/py-tabulate-0.9.0-6xgmssa6olalms4vmt3ttfinvssnq3fh/bin/tabulate
```

**Comment by wdconinc** [comment]
I was hoping to give the above `PATH` concerns a test with a quick build, but it's pulling in `llvm` now... :-(

**Comment by aweaver1fandm** [comment]
I can test it tomorrow morning.  I have a vague recollection that when I
initially started this package I didn't had the path and it didn't run but
I'll test it out to see

**Comment by aweaver1fandm** [comment]
I did some testing and if I do a spack load then yes I can see my
executables and the python used for the build even if I don't include the
path modifications
We use modules though for using software instead of spack load and I am not
sure at the moment if that is working.  The software does spack install but
for some reason
I am currently troubleshooting it will not create a module file

**Comment by wdconinc** [comment]
> We use modules though for using software

I see; I didn't test with modules. I am very unfamiliar with how modules work :-( From what I read, they should however pick up if anything gets installed in the `<prefix>/bin/` directory, and automatically set the PATH for those packages.

**Review by wdconinc — APPROVED** [review]
Looks good.

**Comment by wdconinc** [comment]
You may want to report the issue in your last commit to astropy (if you haven't already done that).

**Comment by alalazo** [comment]
@wdconinc Maybe let's shorten the commit message in future commits. Something like this:

Screenshot from 2024-06-05 18-16-25

with full screen output is very hard to handle on IDEs :slightly_smiling_face:

**Comment by wdconinc** [comment]
Sure thing!

**Pull request merged**